### PR TITLE
Add lsp command to fix rust-analyzer

### PR DESCRIPTION
**Pull request by jackos**
This is continuing from #911, as requested I've moved functionality into a subcommand that can be run like `rustings lsp` and added documentation plus added it to the initial welcome message.

This will ensure a good developer experience for newcomers to the Rust ecosystem.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -367,6 +396,8 @@ started, here's a couple of notes about how Rustlings operates:
 4. If an exercise doesn't make sense to you, feel free to open an issue on GitHub!
    (https://github.com/rust-lang/rustlings/issues/new). We look at every issue,
    and sometimes, other learners do too so you can help each other out!
+5. If you use `rust-analyzer` (provides features like completions) and you want it 
+   to work with exercies run the command `rustlings lsp`
```
Comment (diannasoreil) [inline src/main.rs:400]:
```suggestion
   to work with exercises run the command `rustlings lsp` 
```

File: src/main.rs [hunk 2]
```diff
@@ -367,6 +396,8 @@ started, here's a couple of notes about how Rustlings operates:
 4. If an exercise doesn't make sense to you, feel free to open an issue on GitHub!
    (https://github.com/rust-lang/rustlings/issues/new). We look at every issue,
    and sometimes, other learners do too so you can help each other out!
+5. If you use `rust-analyzer` (provides features like completions) and you want it
```
Comment (diannasoreil) [inline src/main.rs:399]:
```suggestion
5. If you want to use `rust-analyzer` (which provides features like autocompletion) and you want it 
```

**Review by diannasoreil — COMMENTED** [review]
Thank you! Just two little nits.

**Comment by jackos** [comment]
> Thank you! Just two little nits.

Thanks for that, I squashed in the changes

**Comment by diannasoreil** [comment]
@all-contributors please add @jackos for code

**Pull request merged**